### PR TITLE
Introduce main menu on shake gesture

### PR DESCRIPTION
**Pull request by msmoljan**

### CODE REVIEW

**Comment by msmoljan** [comment]
Currently only a menu is put between the shake gesture and translation editing. 

Some refactoring has been done to account for the lifecycle.

**Comment by msmoljan** [comment]
Next steps:
- Fix how lifecycle is handled during translation editing / review
- Implement feedback with screenshots
- Add and finalize corresponding menu options

File: nstack-kotlin-android/src/main/java/dk/nodes/nstack/kotlin/features/common/ActiveActivityHolder.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        private set
+
+    var foregroundActivity: Activity?
+        get() = foregroundActivityReference.get()
+        private set(value) {
+            throw UnsupportedOperationException(
+                    "The foreground activity property should never be explicitly set! " +
+                    "Use foregroundActivityReference to hold on to the activity."
+            )
+        }
```
Comment (lucassales2) [inline nstack-kotlin-android/src/main/java/dk/nodes/nstack/kotlin/features/common/ActiveActivityHolder.kt]: Isn't it easier to just make it a val?

Comment (msmoljan) [reply nstack-kotlin-android/src/main/java/dk/nodes/nstack/kotlin/features/common/ActiveActivityHolder.kt]: It is, but it's an "unclean" API then. To the outside user a `var` represents a variable reference (which this actually is), and a `val` might suggest this doesn't change after the object has been initialized.

Comment (msmoljan) [reply nstack-kotlin-android/src/main/java/dk/nodes/nstack/kotlin/features/common/ActiveActivityHolder.kt]: Simplified :)

**Pull request merged**